core/arch/arm/tee/svc_cache.c

```diff
@@ -60,5 +59,5 @@ TEE_Result syscall_cache_operation(void *va, size_t len,
 	if (res != TEE_SUCCESS)
 		return TEE_ERROR_ACCESS_DENIED;
 
-	return cache_operation(op, va, len);
+	return cache_operation((enum utee_cache_operation)op, va, len);
```

Comment:
strictly speaking, `cache_operation()` expects an ID from enumerated `utee_cache_operation.` Hence i would rather cast here so that the call fulfils the function prototype. I considered  `cache_operation()` as a core internal routine while `syscall_cache_operation()` is the syscall handler, hence conforming the `unsigned long` type used in the ABI.